Added Methods

todorzag opened the pull request.

todorzag committed:
Added Methods

todorzag committed:
Added Objects

mvlt commented on Methods/Problem13/Program.cs:
```diff
@@ -0,0 +1,151 @@
+{
+    internal class Program
+    {
+        static void Main(string[] args)
```
Is all that logic necessary? 

Try having two Read Line for first polynom and next polynom and create (array of the numbers {-5, 1, 3}) like explained in Solutions and Guidelines Section. Try to find some rules and use them in the functions.

todorzag committed:
Fixed Merge

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,151 @@
+﻿namespace Problem11
```
After using option "1" with invalid input - second attempt with valid input return first (invalid) input reversed.

todorzag committed:
Fixed Issues

todorzag committed:
Fixed Problem11

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+
+        public static string ReturnChosenOption()
+        {
+            Console.WriteLine("Option 1: Put the digits from an integer number into a reversed order.");
```
What will you do if you want to add another option?

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+
+            int n = int.Parse(integar);
+
+            if (!(1 <= n && n <= 50000000))
```
Is there more readable version of this logic?

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+            return option;
+        }
+
+        public static string ValidateOption(string option)
```
What will you expect from a function to do with name started with Validate?

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+
+                option = Console.ReadLine();
+
+                option = ValidateOption(option);
```
Why using recursion?

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+            if (split.Contains(" "))
+            {
+                Console.WriteLine("Sequence contains empty elements!");
+                sequence = ValidateSequence();
```
Why using recursion?

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+            Console.WriteLine("Option 2: Calculate the average of given sequence of numbers.");
+            Console.WriteLine("Option 3: Solve the linear equation a * x + b = 0");
+
+            Console.WriteLine("Please enter option number:");
```
Repeated code

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+            {
+                Console.WriteLine("Invalid option!");
+
+                Console.WriteLine("Please enter option number:");
```
Repeated code

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,161 @@
+                integar = ValidateIntegarRange();
+            }
+
+            return integar;
```
IntegAr

mvlt reviewed (commented):
Function should do only one thing

todorzag committed:
Problem11 fixed issues v2

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,176 @@
+﻿namespace Problem11
+{
+    internal class Program
```
Added new comment just to change the status

todorzag committed:
Changes made:
- Made options into Dictionary
- Change all Validate Methods logic to throw error
- Remade Methods to do only one thing

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -21,74 +21,74 @@ static void Main(string[] args)
             };
 
-            foreach (string line in options)
+            foreach (var item in options)
```
Can be wrapped in a function

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -21,74 +21,74 @@ static void Main(string[] args)
             {
-                Console.WriteLine("Invalid option!");
-                return false;
+                throw new ArgumentException("Invalid Option!");
```
Why do we need to throw Exception in validate function? Why do we need to pass the whole dictionary?

mvlt committed:
Merge pull request #4 from todorzag/Objects

Added Objects

todorzag committed:
Changes made:
- Removed throw exceptions
- Added if/else's to work with validate

todorzag committed:
Merge branch 'Methods' of https://github.com/todorzag/from-book into Methods

todorzag committed:
Made previously discussed changes to validate

todorzag committed:
Merge branch 'Methods' of https://github.com/todorzag/from-book into Methods

# Conflicts:
#	Methods/Problem11/Program.cs

todorzag committed:
Changed ValidateOption to IsValidOption

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -37,7 +37,7 @@ static void Main(string[] args)
                 string option = Console.ReadLine();
 
-                if (ValidateOption(option, options))
+                if (IsValidOption(option, options))
```
Function's names should be verb like
use explanatory variable in if statement for better readability

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -28,68 +28,118 @@ static void Main(string[] args)
+                        case "1":
+                            try
+                            {
+                                Console.WriteLine("Please enter integar to reverse:");
```
Each option's handler could be wrapped in separate function

todorzag committed:
Resolved Conflict

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -144,5 +194,21 @@ public static bool ValidateCoefficient(int a)
+            bool result = false;
+
+            Console.WriteLine("Would you like to try again?");
+            string answer = Console.ReadLine();
```
Is there better way of declaring answer when we read from the console

mvlt commented on Methods/Problem11/Program.cs:
```diff
@@ -0,0 +1,214 @@
+        {
+            bool result = true;
+
+            if (!options.ContainsKey(option))
```
Why do we need to reverse the logic in this statement

todorzag committed:
Remade problem13:
Input now 2 ConsoleReadLines
CalculateSum now takes 2 polynomials as args
Removed AddOperator and ReturnPolynomialString

todorzag committed:
Variable declaration
Wrap logic in individual function

todorzag committed:
Created OptionHandler
